src/modules/clustering/clustering.h

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2025 Team Dissolve and contributors
+
+#pragma once
+
+#include "analyser/siteFilter.h"
+#include "analyser/siteSelector.h"
+#include "base/processPool.h"
+#include "classes/configuration.h"
+#include "classes/coreData.h"
+#include "generator/context.h"
+#include "main/dissolve.h"
+#include "module/context.h"
+#include "module/module.h"
+#include <unordered_set>
+
+class Atom;
+
+// Clustering Module
+class ClusteringModule : public Module
+{
+    public:
+    ClusteringModule();
+    ~ClusteringModule() override = default;
+
+    /*
+     * Definition
+     */
+    private:
+    // Target configurations
+    Configuration *targetConfiguration_{nullptr};
+
+    // Simulation data...
+    // Struct
+    struct BondInfo
+    {
+        const SpeciesSite *a_, *b_;
+        double cutOff;
+
+        BondInfo(const SpeciesSite *site1, const SpeciesSite *site2, double cut) : a_(site1), b_(site2), cutOff(cut) {}
+    };
+
```

Comment:
If it was me I would break stuff out of this `BondInfo` structure and just have them as member variables in the module. I get why you added it, but I don't think it's necessary now or in future evolutions of the module TBH.
```suggestion
    // Sites involved in the cluster
    const SpeciesSite *a_, *b_;
    // Maximum allowed distance for two sites to be considered in the same cluster
    double cutoff_{3.0};
```

Reply:
A better example of why this might be useful is for moonshine: looking at clustering due to hydrophobicity would require Methanol-Methanol, Methanol-Ethanol, and Ethanol-Ethanol to be considered all at once for a complete picture. Will certainly rename it though if you're convinced!

Reply:
Coulda sworn I replied to this one this morning at around 10:20, but apparently not! I get why you would want to use a struct, but if you do that you will also need a custom keyword to handle it, a custom keyword widget to display/edit it etc.  In the simple use case (which we should be focusing on) and in the one you've cited if we're talking about hydrogen bond networks then the cutoff is more-or-less going to be the same between all the different OH groups, so one can define general A and B sites containing sites from the different molecules.  The analysis, however, can freely partition its results and output by site type if it wanted to.

If there are more complicated cases where there are two distinctly different types of contacts related to the cluster, this sounds like a module extension or entirely new module.